Assets/_BForBoss/_LevelDesign/Managers/StateManager.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+
+namespace BForBoss
+{
+    public enum State
+    {
+        PreGame,
+        Play,
+        Pause,
+        Death,
```

Comment:
This seems appropriate for now, but I would really like to push for a more robust Statemachine in the future with half step states (i.e. how Unity handles it[ PlayState ](https://docs.unity3d.com/ScriptReference/PlayModeStateChange.html). This is so that we can handle things like Cleanup, Reset and other similar management in the in between states and leave the main states with less clutter work to do

Reply:
yeah for sure, that definitely would be ideal. 

Pros of that - more flexible and fine tuned
Cons of that - it can get bloated and way too much. 

So might need to have sub states as well, or like create a separate life cycle for the substate e.g. the PlayState. Just something to think about